### PR TITLE
Doronst/training set consistency check

### PR DESCRIPTION
**Pull request by doron-st**
Quick-check that training sample mathces expected ground-truth (germline/somatic)
Should be called before make-training-examples

```
gs://ultimagen-runs-data/016576-20221119_0103/016576-NA24385-2-UGAv3-4-CGGCTAGATGCAGAT/016576-NA24385-2-UGAv3-4-CGGCTAGATGCAGAT.cram - target sample match ground truth hit_fraction=0.982468443197756
gs://runs-data-central/011288_20220605_1421/011288-UGAv3-20-CATCCTGCATCGCAGAT/011288-UGAv3-20-CATCCTGCATCGCAGAT.cram - target sample can be normal-in-tumor sample, hit_fraction=0.006311360448807854
gs://ultimagen-sarah/crams/FULL/4_6_50/004777-UGAv3-17-CACATATGGCATATGAT.cram - target sample can be normal-in-tumor sample, hit_fraction=0.005610098176718092
gs://ultimagen-runs-data/020043-20230401_1007/020043-NA12878-2-UGAv3-5-CATCCTGTGCGCATGAT/020043-NA12878-2-UGAv3-5-CATCCTGTGCGCATGAT.cram - target sample can be normal-in-tumor sample, hit_fraction=0.0007012622720897616
gs://ultimagen-runs-data/019160-20230323_1214/019160-UGAv3-5-CATCCTGTGCGCATGAT/019160-UGAv3-5-CATCCTGTGCGCATGAT.cram - target sample can be normal-in-tumor sample, hit_fraction=0.0
gs://runs-data-central/011288_20220605_1421/011288-UGAv3-224-CACAACATATCAGAT/011288-UGAv3-224-CACAACATATCAGAT.cram - normal sample is complementary to ground truth, hit_fraction=0.0
gs://runs-data-central/011288_20220605_1421/011288-UGAv3-28-CGCATCCTCACAGAT/011288-UGAv3-28-CGCATCCTCACAGAT.cram - normal sample is complementary to ground truth, hit_fraction=0.008415147265077139
gs://runs-data-central/011288_20220605_1421/011288-UGAv3-20-CATCCTGCATCGCAGAT/011288-UGAv3-20-CATCCTGCATCGCAGAT.cram - suspected normal-in-tumor sample matches wes_consistency_check/011288-UGAv3-28-CGCATCCTCACAGAT.germline.bed with hit_fraction=1.0
gs://ultimagen-sarah/crams/FULL/4_6_50/004777-UGAv3-17-CACATATGGCATATGAT.cram - suspected normal-in-tumor sample matches wes_consistency_check/011288-UGAv3-28-CGCATCCTCACAGAT.germline.bed with hit_fraction=0.9976174268209667
gs://ultimagen-runs-data/020043-20230401_1007/020043-NA12878-2-UGAv3-5-CATCCTGTGCGCATGAT/020043-NA12878-2-UGAv3-5-CATCCTGTGCGCATGAT.cram - suspected normal-in-tumor sample matches wes_consistency_check/011288-UGAv3-28-CGCATCCTCACAGAT.germline.bed with hit_fraction=0.9996595165134491
gs://ultimagen-runs-data/019160-20230323_1214/019160-UGAv3-5-CATCCTGTGCGCATGAT/019160-UGAv3-5-CATCCTGTGCGCATGAT.cram - suspected normal-in-tumor sample matches wes_consistency_check/011288-UGAv3-28-CGCATCCTCACAGAT.germline.bed with hit_fraction=1.0
```

### CODE REVIEW

File: ugvc/pipelines/deepvariant/training_set_consistency_check.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+            self.sp.print_and_run(
+                f"bedtools intersect -a {self.hcr} -b {self.out_dir}/region.bed | sort -k 1,1 -k 2,2n > {hcr_in_region}"
+            )
+        else:
+            self.sp.print_and_run(f"sort -k 1,1 -k 2,2n {self.hcr} > {hcr_in_region}")
+
+        # Validate that eacg target cram correlates or anti-correlates the ground-truth
+        for target_cram in self.target_crams:
+            hit_fraction = self.calc_hit_fraction(target_cram, ground_truth_to_check)
+            if hit_fraction < 0.95:
```
Comment (ilyasoifer) [inline ugvc/pipelines/deepvariant/training_set_consistency_check.py]: A bit worried about this hardcoded number (recall in low AF CRAMs could be lower)

**Comment by ilyasoifer** [comment]
@doron-st - happen to know why tests fail?

File: ugvc/pipelines/deepvariant/training_set_consistency_check.py (hunk 1 above)
Comment (doron-st) [reply ugvc/pipelines/deepvariant/training_set_consistency_check.py]: Right. I already externalized it after it failed on FFPE

**Comment by doron-st** [comment]
> @doron-st - happen to know why tests fail?

silly typo

**Review by mayalevy — APPROVED** [review]

**Pull request merged**